data/src/main/java/com/fernandocejas/android10/sample/data/entity/mapper/EntityJsonMapper.java

```diff
@@ -15,62 +15,76 @@
  */
 package com.fernandocejas.android10.sample.data.entity.mapper;
 
-import com.fernandocejas.android10.sample.data.entity.UserEntity;
 import com.google.gson.Gson;
 import com.google.gson.JsonSyntaxException;
 import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
 import java.util.List;
-import javax.inject.Inject;
 
 /**
  * Class used to transform from Strings representing json to valid objects.
  */
-public class UserEntityJsonMapper {
+public class EntityJsonMapper<T> {
 
   private final Gson gson;
+  private final Type type;
+  private ParameterizedType parameterizedType;
 
-  @Inject
-  public UserEntityJsonMapper() {
+  public EntityJsonMapper(Class<T> clazz) {
     this.gson = new Gson();
+    this.type = clazz;
   }
 
   /**
-   * Transform from valid json string to {@link UserEntity}.
+   * Transform from valid json string to an object type {@link T}.
    *
-   * @param userJsonResponse A json representing a user profile.
-   * @return {@link UserEntity}.
+   * @param json A json representing an object type {@link T}.
+   * @return {@link T}.
    * @throws com.google.gson.JsonSyntaxException if the json string is not a valid json structure.
    */
-  public UserEntity transformUserEntity(String userJsonResponse) throws JsonSyntaxException {
+  public T transformUserEntity(String json) throws JsonSyntaxException {
```

Comment:
If this is a generic entity mapper, then I think the naming of this methods should reflect it, they can be `transfromFromEntity` and `transfromFromEntityCollection`